File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cs
public class Solution {
    public int[] SearchRange(int[] nums, int target){
        int[] result=new int[2];
        result[0]=FindFirst(nums,target);
        result[1]=FindLast(nums,target);
        return result;
    }
    private int FindFirst(int[] nums, int target){int left=0; int right=nums.Length-1; int index=-1;
    while (left <=right){int mid=left+(right-left)/2;
    if (nums[mid]==target){
        index=mid;
        right=mid-1;
    }else if(nums[mid]<target){left=mid+1;}else{right=mid-1;}}
    return index;}private int FindLast(int[] nums, int target){
        int left=0; int right=nums.Length-1; int index=-1;
        while (left <=right){
            int mid=left+(right-left)/2;
            if(nums[mid]==target){
                index=mid;
                left=mid+1;
            }else if(nums[mid]<target){left=mid+1;}
            else{right=mid-1;}
        }return index;
    }
}

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var searchRange = function(nums, target) {
    let left=findBound(nums, target, true);
    if(left===-1) return [-1,-1];
    let right=findBound(nums,target,false);
    return [left,right];
};
function findBound(nums, target,isLeft){
    let left=0, right=nums.length-1, bound=-1;
    while(left <=right){
        let mid=Math.floor((left+right)/2);
        if(nums[mid]===target){
            bound=mid;
            if(isLeft) right=mid-1;
            else left=mid+1;
        }else if(nums[mid] <target){left=mid+1;}
        else{right=mid-1;}
    }return bound;
};